agent/consul/gateways/bind.go

```diff
@@ -71,3 +84,52 @@ func getReferences(route structs.BoundRoute) (referenceSet, gatewayRefs) {
 
 	return parentRefs, gatewayRefs
 }
+
+func requestToResourceRef(req controller.Request) structs.ResourceReference {
+	ref := structs.ResourceReference{
+		Kind: req.Kind,
+		Name: req.Name,
+	}
+	if req.Meta != nil {
+		ref.EnterpriseMeta = *req.Meta
+	}
+	return ref
+}
+
+// RemoveGateway sets the route status for the given gateway to be unbound if it should be bound
+func RemoveGateway(gateway structs.ResourceReference, entries ...structs.BoundRoute) []structs.ControlledConfigEntry {
+	now := time.Now().UTC()
+	modified := []structs.ControlledConfigEntry{}
+	for _, route := range entries {
+		updater := structs.NewStatusUpdater(route)
+		for _, parent := range route.GetParents() {
+			if parent.Kind == gateway.Kind && parent.Name == gateway.Name && parent.EnterpriseMeta.IsSame(&gateway.EnterpriseMeta) {
+				updater.SetCondition(structs.Condition{
+					Type:               "Bound",
+					Status:             "False",
+					Reason:             "GatewayNotFound",
+					Message:            "gateway was not found",
+					Resource:           &parent,
```

Comment:
will this run into the issue of capturing a pointer to the loop variable `parent` and not the actual value that is there?

Reply:
👍 good catch -- I'll check for other captured range vars as well.